astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+            # Number of digits after decimal is replaced by the precision
+            # for values in Scientific notation, when writing that Format.
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxprec)
+            col.format = str(col.meta.size) + "." + str(maxdec) + "e"
+        else:
+            if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                col.meta.size = maxent + maxdec + 1
+                if sign:
+                    col.meta.size += 1
+            col.fortran_format = fformat + str(col.meta.size) + "." + str(maxdec)
+            col.format = col.fortran_format[1:] + "f"
+
+    def write_byte_by_byte(self):
+        """
+        Writes the Byte-By-Byte description of the table.
+
+        Columns that are `astropy.coordinates.SkyCoord` or `astropy.time.TimeSeries`
+        objects or columns with values that are such objects are recognized as such,
+        and some predefined labels and description is used for them.
+        See the Vizier CDS Standard documentation in the link below for more details
+        on these. An example Byte-By-Byte table is shown here.
+
+        See: http://vizier.u-strasbg.fr/doc/catstd-3.1.htx
+
+        Example::
+
+        --------------------------------------------------------------------------------
+        Byte-by-byte Description of file: table.dat
+        --------------------------------------------------------------------------------
+        Bytes Format Units  Label     Explanations
+        --------------------------------------------------------------------------------
+         1-  8  A8     ---    names   Description of names
+        10- 14  E5.1   ---    e       [-3160000.0/0.01] Description of e
+        16- 23  F8.5   ---    d       [22.25/27.25] Description of d
+        25- 31  E7.1   ---    s       [-9e+34/2.0] Description of s
+        33- 35  I3     ---    i       [-30/67] Description of i
+        37- 39  F3.1   ---    sameF   [5.0/5.0] Description of sameF
+        41- 42  I2     ---    sameI   [20] Description of sameI
+        44- 47  F4.1   h      RAh     Right Ascension (hour)
+        49- 51  F3.1   min    RAm     Right Ascension (minute)
+        53- 70  F18.15 s      RAs     Right Ascension (second)
+            72  A1     ---    DE-     Sign of Declination
+        73- 76  F5.1   deg    DEd     Declination (degree)
+        78- 81  F4.1   arcmin DEm     Declination (arcmin)
+        83- 98  F16.13 arcsec DEs     Declination (arcsec)
+
+        --------------------------------------------------------------------------------
+        """
+        # Get column widths
+        vals_list = []
+        col_str_iters = self.data.str_vals()
+        for vals in zip(*col_str_iters):
+            vals_list.append(vals)
+
+        for i, col in enumerate(self.cols):
+            col.width = max([len(vals[i]) for vals in vals_list])
+            if self.start_line is not None:
```

Comment:
The data part of the CDS table can itself have header lines, esp. when the data is written in a separate file. The title of the Byte-By-Byte for such tables then says that rows till `start_line` are header lines.
Because such tables are not ubiquitous, adding the `start_line` keyword to `ascii.write` for allowing this feature prob skipped my mind.
Should this be added? Else I can remove that `if` statement.

Reply:
I think we leave it in for now.